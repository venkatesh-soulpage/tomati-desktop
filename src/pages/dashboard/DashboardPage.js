import React, { useEffect } from "react";
// redux
import { connect } from "react-redux";
import * as Action from "_actions";
// react-router
import { Route, Switch } from "react-router-dom";
// local components
import SideDrawer from "../SideDrawer";
import Outlet from "../Outlet";
import Event from "../Event";
import ViewOutlet from "../ViewOutlet";
import ViewEvent from "../ViewEvent";
import AddOutlet from "../AddOutlet";
import AddEvent from "../AddEvent";
import Settings from "../Settings";
import AllUsers from "../AllUsers";
import User from "../User";

// Router
import { withRouter } from "react-router-dom";

function DashboardPage(props) {
  useEffect(() => {
    window.scroll(0, 0);
    props.dispatch(Action.getUserData());
  }, []);

  return (
    <div className="container mt-4">
      <div className="mt-100">
        <div className="dashboard-grid-wrapper">
          <div className="dashboard-grid-header"></div>
          <div className="dashboard-left-sidebar">
            <SideDrawer />
          </div>
          <div className="dashboard-grid-main">
            <Switch>
              {/* <Route exact path={props.match.path} component={Outlet} /> */}

              <Route
                exact
                path={`${props.match.path}/outlet`}
                component={Outlet}
              />
              {/* <Route
                exact
                path={`${props.match.path}/event`}
                component={Event}
              /> */}
              <Route
                path={`${props.match.path}/viewoutlet`}
                component={ViewOutlet}
              />
              {/* <Route
                path={`${props.match.path}/viewevent`}
                component={ViewEvent}
              /> */}
              <Route
                exact
                path={`${props.match.path}/addoutlet`}
                component={AddOutlet}
              />
              {/* <Route
                exact
                path={`${props.match.path}/addevent`}
                component={AddEvent}
              /> */}
              <Route
                exact
                path={`${props.match.path}/settings`}
                component={Settings}
              />
              <Route
                exact
                path={`${props.match.path}/all-users`}
                component={AllUsers}
              />
              <Route exact path={`${props.match.path}/user`} component={User} />
            </Switch>
          </div>
        </div>
      </div>
    </div>
  );
}

function mapStateToProps(state) {
  return { auth: state.auth };
}

export default withRouter(connect(mapStateToProps)(DashboardPage));
