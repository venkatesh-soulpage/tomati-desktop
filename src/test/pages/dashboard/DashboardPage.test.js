import React from "react";
import { render } from "@testing-library/react";
import DashboardPage from "pages/dashboard/DashboardPage";
import { Router } from "react-router-dom";
import history from "utils/history";
import { Provider } from "react-redux";
import configureStore from "store";

describe("DashboardPage page", () => {
  it("matches snapshot", () => {
    const store = configureStore();
    const { asFragment } = render(
      <Provider store={store}>
        <Router history={history}>
          <DashboardPage />
        </Router>
      </Provider>,
      {}
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
