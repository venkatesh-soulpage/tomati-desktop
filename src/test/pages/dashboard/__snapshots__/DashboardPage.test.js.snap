// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardPage page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container mt-4"
  >
    <div
      class="mt-100"
    >
      <div
        class="dashboard-grid-wrapper"
      >
        <div
          class="dashboard-grid-header"
        />
        <div
          class="dashboard-left-sidebar"
        >
          <div
            class="d-flex justify-content-center align-items-center"
          >
            <div
              class="spinner-border text-secondary"
              role="status"
            >
              <span
                class="sr-only "
              >
                Loading...
              </span>
            </div>
          </div>
        </div>
        <div
          class="dashboard-grid-main"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
